db/patterns/google_tag_manager.eno

```diff
@@ -6,8 +6,8 @@ alias: google_tag
 
 --- filters
 ||googletagmanager.com/gtm.js^$3p
-||googletagmanager.com/a?^$3p
-||googletagmanager.com/td?^$3p
+||googletagmanager.com/a\?(.+)&gtm=^$3p
+||googletagmanager.com/td?id=GTM-^$3p
```

Comment:
```suggestion
/^https:\/\/www\.googletagmanager\.com\/a\?(?:.+&)?gtm=/$3p
/^https:\/\/www\.googletagmanager\.com\/td\?(?:.+&)?id=GTM-/$3p
```

Notes:
* Switch to regular expression syntax
* Make no assumption about the position of the `gtm` and `id=GTM-` parameters
* Implementation detail: `?:` only to not store the capture group, since we do not access it